website/docs/reference/global-cli-flags.md

```diff
@@ -87,7 +87,9 @@ $ dbt --no-write-json run
 
 ## Strict
 
-The `-S` or `--strict` flag runs schema validations on dbt objects at runtime. This flag may incur a performance penalty, but it is useful for catching logic errors in development of the dbt project.
+The `-S` or `--strict` flag runs schema validations on dbt python objects at runtime. This flag may incur a performance penalty, but it is useful for catching logic errors in development of the dbt project.
+
+**N.B.** In versions >=0.15.0, dbt uses [hologram](https://github.com/fishtown-analytics/hologram) and [mypy](http://mypy-lang.org/) for object type declaration, validation, and testing. The `--strict` flag has no functional use except as an alias for `--warn-error`. We may choose to someday repurpose it.
```

Comment:
There are a couple things strict does:
 - we do an extra `validate` call in some paths when you set `--strict`. This can be really slow as it requires serializing the object to json to validate it.
 - Connections type checks
 - some internal consistency checks during compilation

There's still no reason users should use --strict over --warn-error. We should rename the flags so `--strict` is just an alias for `--warn-error` and `--strict` becomes `--strict-development-checks` or something similarly verbose and obnoxious. You should only run it automatically via integration tests anyway.